Code/Framework/AzToolsFramework/AzToolsFramework/ToolsComponents/EditorComponentBase.cpp

```diff
@@ -49,45 +51,74 @@ namespace AzToolsFramework
             m_transform = nullptr;
         }
 
-        void EditorComponentBase::OnPrepareForAdditionToEntity(AZ::Entity* entity)
+        void EditorComponentBase::OnAfterEntitySet()
         {
-            if (entity && m_alias.empty())
+            AZ_Assert(m_entity, "OnAfterEntitySet - Entity should not be nullptr.");
```

Comment:
is m_entity even used in this function?  Its probably not useful to assert here if thats the case.

Reply:
I think an assert is needed here to ensure `OnAfterEntitySet` is called after `SetEntity`. But I would also add a bail out clause to return early if no entity.